salt/modules/timezone.py

```diff
@@ -410,6 +410,12 @@ def set_hwclock(clock):
         elif clock == 'localtime':
             __salt__['file.sed']('/etc/default/rcS', '^UTC=.*', 'UTC=no')
     elif 'Gentoo' in __grains__['os_family']:
+        if clock not in ('UTC', 'localtime'):
+            raise SaltInvocationError(
+                'Only \'UTC\' and \'localtime\' are allowed'
+            )
+        if clock.__eq__('localtime'):
```

Comment:
Can you explain why you are invoking a dunder method here instead of simply using the ``==`` operator? Especially since invoking the dunder is so much slower:

```python
>>> import timeit
>>> t1 = timeit.Timer('clock == "localtime"', 'clock = "localtime"')
>>> t2 = timeit.Timer('clock.__eq__("localtime")', 'clock = "localtime"')
>>> t1.repeat()
[0.061421871185302734, 0.0285952091217041, 0.026736021041870117]
>>> t2.repeat()
[0.1310901641845703, 0.09536910057067871, 0.09479284286499023]
```